layers/isis.go

```diff
@@ -0,0 +1,486 @@
+	Reserved                   byte
+	MaximumAreaAddresses       byte
+}
+
+type SpecificHeaderType byte
+
+const (
+	IIHL1  SpecificHeaderType = 15
+	IIHL2  SpecificHeaderType = 16
+	IIHP2P SpecificHeaderType = 17
+	L1LSP  SpecificHeaderType = 18
+	L2LSP  SpecificHeaderType = 20
+	L1CSNP SpecificHeaderType = 24
+	L2CSNP SpecificHeaderType = 25
+	L1PSNP SpecificHeaderType = 26
+	L2PSNP SpecificHeaderType = 27
+)
+
+// ///////////////////////////////////////////CLV TYPES
+type CLVCode byte
+
+const (
+	AreaAddresses               CLVCode = 1
+	ISNeighbors                 CLVCode = 2
+	ESNeighbors                 CLVCode = 3
+	PartitionDesignatedLevel2IS CLVCode = 4
+	ISNeighborsMac              CLVCode = 6
+	ISNeighborsSNPA             CLVCode = 7
+	Padding                     CLVCode = 8
+	LspEntries                  CLVCode = 9
+	AuthenticationInfo          CLVCode = 10
+	ProtocolsSupported          CLVCode = 129
+	IpInterfaceAddress          CLVCode = 132
+	Hostname                    CLVCode = 137
+	RestartSignaling            CLVCode = 211
+)
+
+type CLV struct {
+	Code   CLVCode
+	Length byte
+	Value  interface{}
+}
+
+// ///////////////////////////////////////////////
+type ISIS struct {
+	BaseLayer
+	Type                 ISISType
+	CH                   PDUCommonHeader
+	SpecificHeader       interface{}
+	VariableLengthFields []CLV
+}
+
+// ///////////////////////////////////////////////HELLO PKG TYPES FOR SPECIFIC HEADER
+type ISISHelloPkg struct {
+	CircuitType    byte
+	SenderSystemId uint64
+	HoldingTimer   uint16
+	PDULength      uint16
+}
+
+type IIHvL1_L2Lan struct {
+	Base     ISISHelloPkg
+	Priority byte
+
+	DesignatedSystemId struct {
+		SystemId     uint64
+		PseudonodeId byte
+	}
+}
+
+type IIHvP2P struct {
+	Base         ISISHelloPkg
+	LocalCircuit byte
+}
+
+// ///////////////////////////////////////////////LSP PACKETS SPECIFIC HEADER
+
+type LspEntry struct {
+	LspSeqNumber      uint32
+	RemainingLifetime uint16
+	Checksum          uint16
+	Id                LspId
+}
+
+type LspId struct {
+	SystemId     uint64
+	PseudonodeId byte
+	FragmentNum  byte
+}
+
+type ISISLsp struct {
+	PDULength         uint16
+	RemainingLifetime uint16
+
+	Id LspId
+
+	SequenceNumber uint32
+	Checksum       uint16
+
+	PartitionRepair byte
+	Attachment      byte
+	LSDBOverload    byte
+	IsType          byte
+}
+
+type ISISPsnp struct {
+	PDULength uint16
+	SourceId  struct {
+		Id        uint64
+		CircuitId byte
+	}
+}
+
+type ISISCsnp struct {
+	ISISPsnp
+
+	StartLspId LspId
+
+	EndLspId LspId
+}
+
+// ///////////////////////////////////////////////ISIS METHODS
+// String conversions for ISISType
+func (i ISISType) String() string {
+	switch i {
+	case ISISHello:
+		return "Hello"
+	case ISISCompleteSequenceNumberPDU:
+		return "ISISCompleteSequenceNumberPDU"
+	case ISISPartialSequenceNumberPDU:
+		return "ISISPartialSequenceNumberPDU"
+	case ISISLinkStatePDU:
+		return "ISISLinkStatePDU"
+	default:
+		return "No Such ISIS Type"
```

Comment:
For debug purposes in default case its better to return value, which we received 
e.g. `return fmt.Sprintf("Unknown(%d)", k)`